flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -69,7 +69,9 @@ public class FlinkCatalogFactory implements CatalogFactory {
   public static final String HIVE_CONF_DIR = "hive-conf-dir";
   public static final String HADOOP_CONF_DIR = "hadoop-conf-dir";
   public static final String DEFAULT_DATABASE = "default-database";
-  public static final String DEFAULT_DATABASE_NAME = "default";
+  public static final String DEFAULT_DATABASE_NAME = "default_database";
```

Comment:
This is needed for 
```
TestIcebergConnector.testCatalogDatabaseConflictWithFlinkDatabase
```
that unit test drop a database, under this condition:
```
if (!isDefaultDatabaseName()) {
        sql("DROP DATABASE `%s`", databaseName());
      }
```
Now, in this unit test context, there only exists one catalog (`default_catalog`) and one database (`default_database`). So, it is impossible to drop that DB because of https://github.com/apache/flink/pull/23501
Also, I noticed that when you switch to `default_catalog` and you try to get the list of databases, the database returned has name `default_database` and not `default`